### DISCUSSION TITLE
Custom background component, how to?

### DISCUSSION THREAD

**Discussion by jaimeolmo** [xyflow/xyflow#2689] [New Features]
Hello! 

The current background component provides three different variants. My question is, how feasible is it to create a custom background similar to the image provided? Basically, a stripe variant with the potential to use them horizontally and vertically. 

There [is a previous discussion around this](https://github.com/wbkd/react-flow/discussions/2091),  and I know from this other discussion that [I can build a custom background component](https://github.com/wbkd/react-flow/discussions/2534#discussioncomment-4034397), but I haven't found this in the docs. 

![image](https://user-images.githubusercontent.com/9887867/207630847-621f690b-214a-4eea-a024-d6650e631314.png)

**Comment by tpetersons** [comment]
Interested!

I am looking for this too, I've an idea to add custom nodes to resolve this but would be nice to have new options for background e.g. "columns" or "rows".

Hope this is in the future plans!

**Reply by moklick** [reply]
We thought about it but it's very hard to implement a generic background component. I would recommend to have look at the current implementation and adjust it to your needs: https://github.com/wbkd/react-flow/blob/main/packages/background/src/Background.tsx

**Reply by tpetersons** [reply]
@moklick Hmm thanks for the idea and heads up..
Did a quick attempt and works very nicely.

@jaimeolmo I am sure you can reuse this to make rows.

ColumnBackground.tsx
```
import { memo, useRef } from 'react';
import { ReactFlowState, useStore } from 'reactflow';
import shallow from 'zustand/shallow';
import cc from 'classcat';

import { ColumnPattern } from './ColumnPattern';

export type ColumnBackground = {
    width?: number;
    color?: string;
    gap?: number;
    className?: string;
};

const selector = (s: ReactFlowState) => ({ transform: s.transform, patternId: `pattern-${s.rfId}` });

function ColumnBackground({ width = 200, color = '#fff', gap = 20, className = '' }: ColumnBackground) {
    const ref = useRef<SVGSVGElement>(null);

    const { transform, patternId } = useStore(selector, shallow);
    const scaledGap: number = gap * transform[2];
    const columnWidth: number = width * transform[2];

    return (
        <svg
            className={cc(['react-flow__background', className])}
            ref={ref}
            style={{
                position: 'absolute',
                width: '100%',
                height: '100%',
                top: 0,
                left: 0,
            }}
        >
            <pattern
                id={patternId}
                x={transform[0] % scaledGap}
                y={transform[1] % scaledGap}
                width={scaledGap}
                height={scaledGap}
                patternUnits="userSpaceOnUse"
            >
                <ColumnPattern color={color} columnWidth={columnWidth} />
            </pattern>
            <rect x="0" y="0" width="100%" height="100%" fill={`url(#${patternId})`} />
        </svg>
    );
}

export default memo(ColumnBackground);
```

ColumnPattern.tsx
```
type ColumnPatternProps = {
    columnWidth?: number;
    color: string;
};

export function ColumnPattern({ color, columnWidth = 350 }: ColumnPatternProps) {
    return <rect width={columnWidth} height={'100%'} fill={color} />;
}

```

![bg](https://user-images.githubusercontent.com/35035507/208747890-fd9a5d7a-e2bc-4135-a7c6-0fdcf79ebde2.gif)

**Reply by jaimeolmo** [reply]
Thanks @moklick &  @tpetersons for taking your time, it was definitely something that went through my mind, I'm impressed with the quick demo, tomorrow I think I'll have an opportunity to do the row version.  Wow thank you again 🙏
